frontend/lib/location/determine_position.dart

```diff
@@ -95,19 +94,21 @@ Future<RequestedPosition> determinePosition(BuildContext context,
 /// Ensures all preconditions needed to determine the current position.
 /// If needed, location permissions are requested.
 ///
-Future<LocationStatus> checkAndRequestLocationPermission(BuildContext context,
-    {bool requestIfNotGranted = true,
-    rationale =
-        "Erlauben Sie der App Ihren Standort zu benutzen, um Akzeptanzstellen in Ihrer Umgebung anzuzeigen.",
-    Future<void> Function()? onDisableFeature,
-    Future<void> Function()? onEnableFeature}) async {
-  var serviceEnabled = await Geolocator.isLocationServiceEnabled();
+Future<LocationStatus> checkAndRequestLocationPermission(
+  BuildContext context, {
+  bool requestIfNotGranted = true,
+  String rationale =
+      "Erlauben Sie der App Ihren Standort zu benutzen, um Akzeptanzstellen in Ihrer Umgebung anzuzeigen.",
+  Future<void> Function()? onDisableFeature,
+  Future<void> Function()? onEnableFeature,
+}) async {
+  final serviceEnabled = await Geolocator.isLocationServiceEnabled();
   if (!serviceEnabled) {
     if (requestIfNotGranted) {
-      final result = await showDialog(
-          context: context,
-          builder: (context) => const LocationServiceDialog());
-      if (result) {
+      final bool? result =
+          await showDialog<bool>(context: context, builder: (context) => const LocationServiceDialog());
+      // TODO: Is this check correct? Both buttons in LocationServiceDialog return true
+      if (result == true) {
```

Comment:
Is this correct @maxammann? Same check is used somewhere below

Reply:
The code here is correct. The only question is why LocationServiceDialog return true for both actions.

Reply:
Do you want to have a look or should I open an issue?

Reply:
Fixed this like suggested below